### PR TITLE
Add documentation for dynamic configuration of tenancy in Dashboards

### LINKED ISSUE #2562: [DOC] Document new multi-tenancy UI for management and configuration
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.**
Configuration for multi-tenancy is currently handled in the config.yml and opensearch_dashboards.yml files. The UX team is replacing the Tenants page in Dashboards with a new Multi-tenancy page that will begin to provide functionality for administrators in both managing and configuring tenants. Documentation needs to provide configuration and management information for both environments: config files and API; Dashboards user interface.

New documentation will provide information about new behavior in the UI and tasks administrators can perform, including:

- Enabling / disabling Multi-tenancy as a feature.
- Disabling the Private tenant.
- Setting a default tenant for all users (choosing between Global or Private).
- Notifications sent to non-admin and admin users about changes to tenant configuration that have an impact on them.

OpenSearch Dashboards multi-tenancy documentation already includes a "Multi-tenancy configuration" topic. We can add relevant documentation here. However, there may be additional changes to the API to align functionality between the two environments, which could require updates in other areas of Security documentation.
 

**What other resources are available?**
RFC for this feature: Issue #[5853](https://github.com/opensearch-project/OpenSearch/issues/5853). 
Some UX-related items:[ #1319](https://github.com/opensearch-project/security-dashboards-plugin/issues/1319).
Multi-tenancy configuration
https://opensearch.org/docs/latest/security/multi-tenancy/multi-tenancy-config/

### PR DESCRIPTION
**Pull request by cwillum**
### Description
This adds documentation for dynamic configuration of certain tenancy features including enabling or disabling tenancy, disabling or enabling private tenants, and setting the default tenant for all users. The documentation also includes new APIs that provide another option for configuring the same dynamic settings.

### Issues Resolved
Fixes #2562 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _security/multi-tenancy/dynamic-config.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  
+  This setting does not have an impact on the global tenant, which always remains enabled.
+  {: .note }
+
+- **Disable or enable private tenant** - This option allows administrators to disable and enable private tenants. As with the enable multi-tenancy setting, when private tenants are re-enabled all previously saved objects are preserved and made available.
+- **Default tenant** - This option allows an administrator to choose either a global, private, or custom tenant as the default when users sign on. In the case where a user doesn't have access to the default tenant (for example, if a custom tenant unavailable to the user was specified as the default), the default transitions to the preferred tenant, which is specified by the `opensearch_security.multitenancy.tenants.preferred` setting in the `opensearch-dashboards.yml` file. See [Multi-tenancy configuration]({{site.url}}{{site.baseurl}}/security/multi-tenancy/multi-tenancy-config/) for details about this setting.
+
+Once you make changes to multi-tenancy with dynamic configuration and save, users are logged out of their sessions and must log back in.
```
Comment (abhivka7) [inline _security/multi-tenancy/dynamic-config.md:21]: This case is conditional. For eg: 
1. If admin disables multi-tenancy, users are not logged out if they are in Global tenant.
2. If admin disables private tenant, users are logged out only if they were in Private tenant.
3. Users are not logged out if only default tenant is changed.

File: _security/multi-tenancy/dynamic-config.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+To make settings for multi-tenancy in Dashboards, Begin by selecting **Security** in the Dashboards home page menu. Then select **Tenancy** from the Security menu on the left side of the screen. The **Multi-tenancy** page is displayed. 
+
+By default, the **Manage** tab is displayed. Select the **Configure** tab to display the dynamic settings for multi-tenancy.
+
+* In the **Multi-tenancy** field, select the **Enable tenancy** check box to enable multi-tenancy. Clear the check box to disable the feature. The default is `true`.
+* In the **Tenants** field, you can enable or disable private tenants for users. By default the check box is selected and the feature is enabled.
+* In the **Default tenant** field, use the dropdown menu to select a default tenant. The menu includes Global, Private, and any other custom tenants that are available to users.
+
+After making selections, select **Save changes** in the lower right corner of the window. The changes are implemented dynamically.
```
Comment (abhivka7) [inline _security/multi-tenancy/dynamic-config.md:34]: After users click Save Changes, they will be asked to confirm whether they are sure about these changes in a form of checkbox modal.

File: _security/multi-tenancy/mt-agg-view.md [hunk 3]
```diff
@@ -36,7 +36,10 @@ In this first experimental phase of development, there are some limitations that
 
 * The feature can only be used in a new cluster. At this time, the feature is not suported by clusters already in use.
 * Also, the feature should be used only in a test environment, not in production. 
-* Finally, once the feature has been enabled and used in a test cluster, the feature cannot be disabled for the cluster. Disabling the feature once it has been used to work with tenants and saved objects can result in the loss of saved objects and have an impact on tenant-to-tenant functionality. 
+* Finally, once the feature has been enabled and used in a test cluster, the feature cannot be disabled for the cluster. Disabling the feature once it has been used to work with tenants and saved objects can result in the loss of saved objects and can have an impact on tenant-to-tenant functionality.
+
+  While aggregate view is enabled, a loss of saved objects and impact to tenant-to-tenant functionality can also occur when the larger multi-tenancy feature is disabled. This includes disabling multi-tenancy in the security `config.yml` file settings or by using dynamic configuration. Keep this in mind before enabling aggregate view for your test cluster.
```
Comment (abhivka7) [inline _security/multi-tenancy/mt-agg-view.md:41]: @RyanL1997 I don't think there will be a loss of saved objects with respect to Aggregation view feature when multi-tenancy is disabled. Can you confirm?

Comment (cwillum) [reply _security/multi-tenancy/mt-agg-view.md:41]: @abhivka7 @RyanL1997 I understood that some changes needed to be made so that aggregate view was compatible with dynamic configuration. The result of those changes was that disabling multi-tenancy would also disable the aggregate view feature. And once both were disabled, it made the last bullet above a concern:
"Finally, once the feature has been enabled and used in a test cluster, the feature cannot be disabled for the cluster. ..."
Let's clarify this so I can make changes to the warning note if necessary. Thanks.

Comment (abhivka7) [reply _security/multi-tenancy/mt-agg-view.md:41]: @cwillum Basically aggregation view is only shown when multitenancy is enabled. If we disable multitenancy then there is no point in showing aggregation view. So aggregation view as a feature in itself can not be disabled once it has been enabled. But it still needs multitenancy to be enabled as a pre-requisite.

File: _security/multi-tenancy/dynamic-config.md (hunk 1 above)
Comment (cwillum) [reply _security/multi-tenancy/dynamic-config.md]: Thanks for the clarification. I've reworded this.

File: _security/multi-tenancy/dynamic-config.md (hunk 2 above)
Comment (cwillum) [reply _security/multi-tenancy/dynamic-config.md]: Thanks for pointing this out. I've reformatted this section into formal steps, including the missing one you mention here.

File: _security/multi-tenancy/mt-agg-view.md [hunk 4]
```diff
@@ -36,7 +36,7 @@ In this first experimental phase of development, there are some limitations that
...
-* Finally, once the feature has been enabled and used in a test cluster, the feature cannot be disabled for the cluster. Disabling the feature once it has been used to work with tenants and saved objects can result in the loss of saved objects and have an impact on tenant-to-tenant functionality. 
+* Finally, once the feature has been enabled and used in a test cluster, the feature cannot be disabled for the cluster. Disabling the feature once it has been used to work with tenants and saved objects can result in the loss of saved objects and can have an impact on tenant-to-tenant functionality. This can occur when disabling the feature in any one of three ways: disabling the aggregate view feature with the [feature flag]({{site.url}}{{site.baseurl}}/security/multi-tenancy/mt-agg-view/#enabling-aggregate-view-for-saved-objects/); disabling multi-tenancy with the traditional [multi-tenancy configuration]({{site.url}}{{site.baseurl}}/security/multi-tenancy/multi-tenancy-config/) setting; or disabling multi-tenancy with [dynamic configuration]({{site.url}}{{site.baseurl}}/security/multi-tenancy/dynamic-config/) settings.
```
Comment (RyanL1997) [inline _security/multi-tenancy/mt-agg-view.md:39]: This is correct! Thanks for summarizing this up! @cwillum

**Review by RyanL1997 — APPROVED** [review]
Hi @cwillum, thanks for putting this together! Your summary of these potential behaviors is clear and accurate.

File: _security/multi-tenancy/mt-agg-view.md (hunk 3 above)
Comment (RyanL1997) [reply _security/multi-tenancy/mt-agg-view.md]: Hi @abhivka7 and @cwillum . I will just move some of our conversation over here for references:
1. This potential risk / known limitation existed since the launch of the aggregation view feature, referencing to the previous documentation here (https://opensearch.org/docs/latest/security/multi-tenancy/mt-agg-view/#known-limitations): 
> Finally, once the feature has been enabled and used in a test cluster, the feature cannot be disabled for the cluster. Disabling the feature once it has been used to work with tenants and saved objects can result in the loss of saved objects and have an impact on tenant-to-tenant functionality.
2. According to the new feature of Dynamically Configurable Multi-tenancy, it didn't change any previous technical design of both aggregation view feature and tenancy feature. However, it does change some accessibility of multi-tenancy configuration. That's reason of @cwillum  is helping us to summarizing and emphasizing some specific details of these potential user behaviors, so that we can warn user about the above risks even with a non-production environment

All in all, big thanks to both of you for the follow-ups. I think we have sorted everything clear.

File: _security/multi-tenancy/dynamic-config.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+layout: default
+title: Dynamic configuration in OpenSearch Dashboards
+parent: OpenSearch Dashboards multi-tenancy
+nav_order: 147
+---
+
+
+# Dynamic configuration in OpenSearch Dashboards
+
+Dynamic configuration of multi-tenancy in OpenSearch Dashboards provides options to make common settings for tenancy without having to make changes to the configuration YAML files on each node and then restart the cluster. You can take advantage of this functionality by using the Dashboards interface or the REST API. The following list includes description of the options currently covered by dynamic configuration:
```
Comment (hdhalter) [inline _security/multi-tenancy/dynamic-config.md:11]: Can we avoid making this into a feature by calling out the functionality? For example, "You can dynamically configure your multi-tenancy settings in OpenSearch Dashboards without making changes to the configuration YAML files on each note and then restarting the cluster."

File: _security/multi-tenancy/dynamic-config.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+Depending on the specific changes made to multi-tenancy using dynamic configuration, some users may be logged out of their Dashboards session once the changes are saved. For example, if an admin user disables multi-tenancy, users with either a private or custom tenant as their selected tenant will be logged out and will need to log back in. Similarly, if an admin user disables private tenants, users with the private tenant selected will be logged out and will need to log back in. 
+
+The global tenant, however, is a special case. Since this tenant is never disabled, users with global tenant selected as their active tenant will experience no interruption to their session. Furthermore, changing the default tenant has no impact on a user's session.
+
+
+## Configuring multi-tenancy in OpenSearch Dashboards
+
+To make settings for multi-tenancy in Dashboards, follow these steps.
```
Comment (hdhalter) [inline _security/multi-tenancy/dynamic-config.md:28]: To configure multi-tenancy... ?

**Review by hdhalter — APPROVED** [review]
Looks good, Chris, with a couple of suggestions. Thanks!

**Review by ariamarble — APPROVED** [review]

File: _security/multi-tenancy/dynamic-config.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+layout: default
+title: Dynamic configuration in OpenSearch Dashboards
+parent: OpenSearch Dashboards multi-tenancy
+nav_order: 147
+---
+
+
+# Dynamic configuration in OpenSearch Dashboards
+
+Multi-tenancy includes dynamic configuration options in OpenSearch Dashboards so you can make common settings for tenancy without having to make changes to the configuration YAML files on each node and then restart the cluster. You can take advantage of this functionality by using the Dashboards interface or the REST API. The following list includes description of the options currently covered by dynamic configuration:
```
Comment (natebower) [inline _security/multi-tenancy/dynamic-config.md:11]:
```suggestion
Multi-tenancy includes dynamic configuration options in OpenSearch Dashboards so you can make common settings for tenancy without having to make changes to the configuration YAML files on each node and then restart the cluster. You can take advantage of this functionality by using the Dashboards interface or the REST API. The following list includes descriptions of the options currently covered by dynamic configuration:
```

Comment (natebower) [inline _security/multi-tenancy/dynamic-config.md:11]: "create common settings" instead of "make"?

File: _security/multi-tenancy/dynamic-config.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+- **Disable or enable multi-tenancy** - Administrators can disable and enable multi-tenancy dynamically. Disabling multi-tenancy does not pose a risk for loss of data. If and when an administrator chooses to re-enable tenancy, all previously saved objects are preserved and made available.  The default is `true`.
```
Comment (natebower) [inline _security/multi-tenancy/dynamic-config.md:13]:
```suggestion
- **Disable or enable multi-tenancy** -- Administrators can disable and enable multi-tenancy dynamically. Disabling multi-tenancy does not pose a risk of data loss. If and when an administrator chooses to reenable tenancy, all previously saved objects are preserved and made available. The default is `true`.
```

File: _security/multi-tenancy/dynamic-config.md [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+  
+  This setting does not have an impact on the global tenant, which always remains enabled.
+  {: .note }
+
+- **Disable or enable private tenant** - This option allows administrators to disable and enable private tenants. As with the enable multi-tenancy setting, when private tenants are re-enabled all previously saved objects are preserved and made available.
```
Comment (natebower) [inline _security/multi-tenancy/dynamic-config.md:18]:
```suggestion
- **Disable or enable private tenant** -- This option allows administrators to disable and enable private tenants. As with the `enable multi-tenancy` setting, when private tenants are reenabled, all previously saved objects are preserved and made available.
```

File: _security/multi-tenancy/dynamic-config.md [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+- **Default tenant** - This option allows an administrator to choose either a global, private, or custom tenant as the default when users sign on. In the case where a user doesn't have access to the default tenant (for example, if a custom tenant unavailable to the user was specified as the default), the default transitions to the preferred tenant, which is specified by the `opensearch_security.multitenancy.tenants.preferred` setting in the `opensearch-dashboards.yml` file. See [Multi-tenancy configuration]({{site.url}}{{site.baseurl}}/security/multi-tenancy/multi-tenancy-config/) for details about this setting.
```
Comment (natebower) [inline _security/multi-tenancy/dynamic-config.md:19]:
```suggestion
- **Default tenant** -- This option allows an administrator to choose either a global, private, or custom tenant as the default when users sign on. In cases where a user doesn't have access to the default tenant (for example, if a custom tenant unavailable to the user was specified as the default), the default transitions to the preferred tenant, which is specified by the `opensearch_security.multitenancy.tenants.preferred` setting in the `opensearch-dashboards.yml` file. See [Multi-tenancy configuration]({{site.url}}{{site.baseurl}}/security/multi-tenancy/multi-tenancy-config/) for more information about this setting.
```

File: _security/multi-tenancy/dynamic-config.md [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+
+Depending on the specific changes made to multi-tenancy using dynamic configuration, some users may be logged out of their Dashboards session once the changes are saved. For example, if an admin user disables multi-tenancy, users with either a private or custom tenant as their selected tenant will be logged out and will need to log back in. Similarly, if an admin user disables private tenants, users with the private tenant selected will be logged out and will need to log back in.
```
Comment (natebower) [inline _security/multi-tenancy/dynamic-config.md:21]: We use "sign on" in the para above but "log in" here.

File: _security/multi-tenancy/dynamic-config.md [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+
+Depending on the specific changes made to multi-tenancy using dynamic configuration, some users may be logged out of their Dashboards session once the changes are saved. For example, if an admin user disables multi-tenancy, users with either a private or custom tenant as their selected tenant will be logged out and will need to log back in. Similarly, if an admin user disables private tenants, users with the private tenant selected will be logged out and will need to log back in. 
+
+The global tenant, however, is a special case. Since this tenant is never disabled, users with global tenant selected as their active tenant will experience no interruption to their session. Furthermore, changing the default tenant has no impact on a user's session.
```
Comment (natebower) [inline _security/multi-tenancy/dynamic-config.md:23]:
```suggestion
The global tenant, however, is a special case. Because this tenant is never disabled, users with the global tenant selected as their active tenant will experience no interruption to their session. Furthermore, changing the default tenant has no impact on a user's session.
```

File: _security/multi-tenancy/dynamic-config.md [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+
+
+## Configuring multi-tenancy in OpenSearch Dashboards
+
+To configure multi-tenancy in Dashboards, follow these steps.
```
Comment (natebower) [inline _security/multi-tenancy/dynamic-config.md:28]:
```suggestion
To configure multi-tenancy in Dashboards, follow these steps: 
```

File: _security/multi-tenancy/dynamic-config.md [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+The global tenant, however, is a special case. Since this tenant is never disabled, users with global tenant selected as their active tenant will experience no interruption to their session. Furthermore, changing the default tenant has no impact on a user's session.
+
+
+## Configuring multi-tenancy in OpenSearch Dashboards
+
+To configure multi-tenancy in Dashboards, follow these steps. 
+
+1. Begin by selecting **Security** in the Dashboards home page menu. Then select **Tenancy** from the Security menu on the left side of the screen. The **Multi-tenancy** page is displayed. 
+1. By default, the **Manage** tab is displayed. Select the **Configure** tab to display the dynamic settings for multi-tenancy.
+   * In the **Multi-tenancy** field, select the **Enable tenancy** checkbox to enable multi-tenancy. Clear the checkbox to disable the feature. The default is `true`.
```
Comment (natebower) [inline _security/multi-tenancy/dynamic-config.md:32]:
```suggestion
   * In the **Multi-tenancy** field, select the **Enable tenancy** check box to enable multi-tenancy. Clear the check box to disable the feature. The default is `true`.
```

File: _security/multi-tenancy/dynamic-config.md [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+   * In the **Tenants** field, you can enable or disable private tenants for users. By default the checkbox is selected and the feature is enabled.
```
Comment (natebower) [inline _security/multi-tenancy/dynamic-config.md:33]:
```suggestion
   * In the **Tenants** field, you can enable or disable private tenants for users. By default, the check box is selected and the feature is enabled.
```

File: _security/multi-tenancy/dynamic-config.md [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+   * In the **Default tenant** field, use the dropdown menu to select a default tenant. The menu includes Global, Private, and any other custom tenants that are available to users.
+1. After making your preferred changes, select **Save changes** in the lower right corner of the window. A popup window appears listing the configuration items you've changed and asks you to review your changes.
```
Comment (natebower) [inline _security/multi-tenancy/dynamic-config.md:35]:
```suggestion
1. After making your preferred changes, select **Save changes** in the lower-right corner of the window. A pop-up window appears listing the configuration items you've changed and asks you to review your changes.
```

File: _security/multi-tenancy/dynamic-config.md [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+1. Select the checkboxes beside the items you want to confirm and then select **Apply changes**. The changes are implemented dynamically.
```
Comment (natebower) [inline _security/multi-tenancy/dynamic-config.md:36]:
```suggestion
1. Select the check boxes beside the items you want to confirm and then select **Apply changes**. The changes are implemented dynamically.
```

File: _security/multi-tenancy/dynamic-config.md [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+
+
+## Configuring multi-tenancy with the REST API
+
+In addition to the Dashboards interface, dynamic configurations can be made using the REST API.
```
Comment (natebower) [inline _security/multi-tenancy/dynamic-config.md:41]: "created" instead of "made"?

File: _security/multi-tenancy/dynamic-config.md [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+1. Select the checkboxes beside the items you want to confirm and then select **Apply changes**. The changes are implemented dynamically.
+
+
+## Configuring multi-tenancy with the REST API
+
+In addition to the Dashboards interface, dynamic configurations can be made using the REST API. 
+
+### Get tenancy configuration
+
+Retrieves settings for the dynamic configuration.
```
Comment (natebower) [inline _security/multi-tenancy/dynamic-config.md:45]:
```suggestion
Retrieves settings for the dynamic configuration:
```

Comment (natebower) [inline _security/multi-tenancy/dynamic-config.md:45]: Can we rephrase this as a complete sentence?

File: _security/multi-tenancy/dynamic-config.md [hunk 20]
```diff
@@ -0,0 +1,109 @@
...
+{
+    "mulitenancy_enabled": true,
+    "private_tenant_enabled": true,
+    "default_tenant": "global tenant"
+}
+```
+
+### Update tenant configuration
+
+Updates settings for dynamic configuration.
```
Comment (natebower) [inline _security/multi-tenancy/dynamic-config.md:64]:
```suggestion
Updates settings for dynamic configuration:
```

Comment (natebower) [inline _security/multi-tenancy/dynamic-config.md:64]: Can we rephrase this as a complete sentence?

File: _security/multi-tenancy/dynamic-config.md [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+{
+    "mulitenancy_enabled": true,
+    "private_tenant_enabled": false,
+    "default_tenant": "custom tenant 1"
+}
+```
+
+### Dashboardsinfo API
+
+You can also use the Dashboardsinfo API to retrieve the status of multi-tenancy settings for the user logged in to Dashboards.
```
Comment (natebower) [inline _security/multi-tenancy/dynamic-config.md:88]:
```suggestion
You can also use the Dashboardsinfo API to retrieve the status of multi-tenancy settings for the user logged in to Dashboards:
```

File: _security/multi-tenancy/multi-tenancy-config.md [hunk 22]
```diff
@@ -15,19 +15,23 @@ config:
...
-`index` | Must match the name of the OpenSearch Dashboards index from `opensearch_dashboards.yml`. Default is `.kibana`.
-`do_not_fail_on_forbidden` | If true, the security plugin removes any content that a user is not allowed to see from search results. If false, the plugin returns a security exception. Default is false.
+| Setting | Description |
+| :--- | :--- |
+| `multitenancy_enabled` | Enable or disable multi-tenancy. Default is true. |
+| `private_tenant_enabled` | Enable or disable the private tenant. Default is true. |
+| `default_tenant` | Use to set the tenant that is available when users log in. |
+| `server_username` | Must match the name of the OpenSearch Dashboards server user from `opensearch_dashboards.yml`. Default is `kibanaserver`. |
+| `index` | Must match the name of the OpenSearch Dashboards index from `opensearch_dashboards.yml`. Default is `.kibana`. |
+| `do_not_fail_on_forbidden` | If true, the security plugin removes any content that a user is not allowed to see from search results. If false, the plugin returns a security exception. Default is false. |
```
Comment (natebower) [inline _security/multi-tenancy/multi-tenancy-config.md:32]:
```suggestion
| `do_not_fail_on_forbidden` | If true, the Security plugin removes any content that a user is not allowed to see from the search results. If false, the plugin returns a security exception. Default is false. |
```

File: _security/multi-tenancy/multi-tenancy-config.md [hunk 23]
```diff
@@ -40,14 +44,14 @@ opensearch_security.multitenancy.tenants.preferred: ["Private", "Global"]
...
+| Setting | Description |
+| :--- | :--- |
+| `opensearch.requestHeadersAllowlist` | OpenSearch Dashboards requires that you add all HTTP headers to the allow list so that the headers pass to OpenSearch. Multi-tenancy uses a specific header, `securitytenant`, that must be present with the standard `Authorization` header. If the `securitytenant` header is not on the allow list, OpenSearch Dashboards starts with a red status.
+| `opensearch_security.multitenancy.enabled` | Enables or disables multi-tenancy in OpenSearch Dashboards. Default is true. |
+| `opensearch_security.multitenancy.tenants.enable_global` | Enables or disables the global tenant. Default is true. |
+| `opensearch_security.multitenancy.tenants.enable_private` | Enables or disables the private tenant. Default is true. |
+| `opensearch_security.multitenancy.tenants.preferred` | Lets you change ordering in the **Tenants** tab of OpenSearch Dashboards. By default, the list starts with global and private (if enabled) and then proceeds alphabetically. You can add tenants here to move them to the top of the list. |
```
Comment (natebower) [inline _security/multi-tenancy/multi-tenancy-config.md:53]: Should "global" and "private" be capitalized or in quotes? They're capitalized in the first file.

File: _security/multi-tenancy/tenant-index.md [hunk 24]
```diff
@@ -10,18 +10,19 @@ redirect_from:
 
 # OpenSearch Dashboards multi-tenancy
 
-*Tenants* in OpenSearch Dashboards are spaces for saving index patterns, visualizations, dashboards, and other OpenSearch Dashboards objects. Tenants are useful for safely sharing your work with other OpenSearch Dashboards users. You can control which roles have access to a tenant and whether those roles have read or write access. By default, all OpenSearch Dashboards users have access to two independent tenants:
+*Tenants* in OpenSearch Dashboards are spaces for saving index patterns, visualizations, dashboards, and other OpenSearch Dashboards objects. OpenSearch allows users to create multiple tenants for multiple uses. Tenants are useful for safely sharing your work with other OpenSearch Dashboards users. You can control which roles have access to a tenant and whether those roles have read or write access. By default, all OpenSearch Dashboards users have access to two independent tenants: private and global. Multi-tenancy also provides the option to create custom tenants.
```
Comment (natebower) [inline _security/multi-tenancy/tenant-index.md:13]: Same comment re: "private" and "global".

File: _security/multi-tenancy/tenant-index.md [hunk 25]
```diff
@@ -10,18 +10,19 @@ redirect_from:
...
 
-- **Private** - This tenant is exclusive to each user and can't be shared. You can't use it to access routes or index patterns made by the user's global tenant.
-- **Global** - This tenant is shared between every OpenSearch Dashboards user. 
+- **Private** - This tenant is exclusive to each user and can't be shared. It does not allow you to access routes or index patterns made by the user's global tenant.
```
Comment (natebower) [inline _security/multi-tenancy/tenant-index.md:15]:
```suggestion
- **Private** -- This tenant is exclusive to each user and can't be shared. It does not allow you to access routes or index patterns made by the user's global tenant.
```

Comment (natebower) [inline _security/multi-tenancy/tenant-index.md:15]: "created" instead of "made"?

File: _security/multi-tenancy/tenant-index.md [hunk 26]
```diff
@@ -10,18 +10,19 @@ redirect_from:
...
+- **Global** - This tenant is shared between every OpenSearch Dashboards user. It does allow for sharing objects among users who have access to it.
```
Comment (natebower) [inline _security/multi-tenancy/tenant-index.md:16]:
```suggestion
- **Global** -- This tenant is shared between every OpenSearch Dashboards user. It does allow for sharing objects among users who have access to it.
```

File: _security/multi-tenancy/tenant-index.md [hunk 27]
```diff
@@ -10,18 +10,19 @@ redirect_from:
...
+- **Custom** - Administrators can create custom tenants and assign them to specific roles. Once created, these tenants can then provide spaces for specific groups of users.
```
Comment (natebower) [inline _security/multi-tenancy/tenant-index.md:17]:
```suggestion
- **Custom** -- Administrators can create custom tenants and assign them to specific roles. Once created, these tenants can then provide spaces for specific groups of users.
```

File: _security/multi-tenancy/tenant-index.md [hunk 28]
```diff
@@ -10,18 +10,19 @@ redirect_from:
...
 
-The global tenant is not a *primary* tenant such that any action done within the global tenant is not replicated to a user's private tenant. If you make a change to your global tenant, you won't see that change reflected in your private tenant. Some example changes include, but are not limited to:
+The global tenant is not a *primary* tenant in the sense that it replicates its content in a private tenant. To the contrary, if you make a change to your global tenant, you won't see that change reflected in your private tenant. Some example changes include, but are not limited to:
```
Comment (natebower) [inline _security/multi-tenancy/tenant-index.md:19]:
```suggestion
The global tenant is not a *primary* tenant in the sense that it replicates its content in a private tenant. To the contrary, if you make a change to your global tenant, you won't see that change reflected in your private tenant. Some example changes include the following:
```

File: _security/multi-tenancy/tenant-index.md [hunk 29]
```diff
@@ -31,5 +32,7 @@ http://<opensearch_dashboards_host>:5601/app/opensearch-dashboards?security_tena
 
 ## Next steps
 
-To get started with tenants, see [Multi-tenancy configuration]({{site.url}}{{site.baseurl}}/security/multi-tenancy/multi-tenancy-config/) for information on enabling multi-tenancy, adding tenants, and assigning roles to tenants.   
+To get started with tenants, see [Multi-tenancy configuration]({{site.url}}{{site.baseurl}}/security/multi-tenancy/multi-tenancy-config/) for information about enabling multi-tenancy, adding tenants, and assigning roles to tenants.
+
+To see information about making dynamic changes to the multi-tenancy configuration, see [Dynamic configuration in OpenSearch Dashboards]({{site.url}}{{site.baseurl}}/security/multi-tenancy/dynamic-config/).
```
Comment (natebower) [inline _security/multi-tenancy/tenant-index.md:37]:
```suggestion
For information about making dynamic changes to the multi-tenancy configuration, see [Dynamic configuration in OpenSearch Dashboards]({{site.url}}{{site.baseurl}}/security/multi-tenancy/dynamic-config/).
```

File: _security/multi-tenancy/multi-tenancy-config.md [hunk 30]
```diff
@@ -15,19 +15,23 @@ config:
...
 
-Setting | Description
-:--- | :---
-`multitenancy_enabled` | Enable or disable multi-tenancy. Default is true.
-`server_username` | Must match the name of the OpenSearch Dashboards server user from `opensearch_dashboards.yml`. Default is `kibanaserver`.
-`index` | Must match the name of the OpenSearch Dashboards index from `opensearch_dashboards.yml`. Default is `.kibana`.
-`do_not_fail_on_forbidden` | If true, the security plugin removes any content that a user is not allowed to see from search results. If false, the plugin returns a security exception. Default is false.
+| Setting | Description |
+| :--- | :--- |
+| `multitenancy_enabled` | Enable or disable multi-tenancy. Default is true. |
```
Comment (natebower) [inline _security/multi-tenancy/multi-tenancy-config.md:27]:
```suggestion
| `multitenancy_enabled` | Enable or disable multi-tenancy. Default is `true`. |
```

Comment (natebower) [inline _security/multi-tenancy/multi-tenancy-config.md:27]: Global: Put all of these default settings, in both tables, in code format, as I've done here.

**Review by natebower — COMMENTED** [review]
@cwillum Please see my comments and changes and let me know if you have any questions. Thanks!

File: _security/multi-tenancy/tenant-index.md (hunk 24 above)
Comment (cwillum) [reply _security/multi-tenancy/tenant-index.md]: I don't so in this case. First, I don't think the global tenant needs to be a proper noun. Where it's referred to as a setting, and it's capitalized in the UI or a key, sure. Likewise with private tenants. Every user gets one. So there's no monolithic private tenant. Just a bunch of separate ones. 
Still, I've reworded this like so:
"... all OpenSearch Dashboards users have access to two independent tenants: the global tenant and a private tenant."

**Review by cwillum — COMMENTED** [review]
Thanks for the quick review.
